### PR TITLE
Allow authentication using cached JSON data

### PR DESCRIPTION
**Pull request by fbis251**

### CODE REVIEW

**Comment by ccrama** [comment]
Is this the file from my JRAW fork? I was gonna put this in one big PR but if you'd like to that's cool :)

**Comment by fbis251** [comment]
Hey, I cherry picked the commit you made with regard to caching authentication data. I changed a few things to that method after some testing, then I also added another method that allows you to use cached JSON data doing userless and app-only authentication.

Although your commit is included in the pull request there are also some more changes in the request building and documentation

**Comment by ccrama** [comment]
Gotcha, looks good!

File: src/main/java/net/dean/jraw/http/oauth/OAuthHelper.java [hunk 1]
```diff
@@ -2,40 +2,42 @@
...
 import java.util.Map;
 
 /**
  * <p>This class assists developers using this library in authenticating users via OAuth2. For the app types of
- *    'installed' or 'web', a typical use of this class is as follows:
- *
+ * 'installed' or 'web', a typical use of this class is as follows:
+ * <p>
  * <ol>
- *     <li>Obtain an authorization URL using {@link #getAuthorizationUrl(Credentials, boolean, String...)}.
```
Comment (thatJavaNerd) [inline src/main/java/net/dean/jraw/http/oauth/OAuthHelper.java]: Please revert documentation formatting

**Comment by fbis251** [comment]
Hey @thatJavaNerd is the last commit enough or should I revert the changes to spacing in the javadoc parameters and descriptions?

**Comment by thatJavaNerd** [comment]
Please revert the formatting before I merge the pull request.

**Comment by fbis251** [comment]
I have submitted a new commit with the changes you requested

**Comment by thatJavaNerd** [comment]
Very nice, thank you both!

**Pull request merged**